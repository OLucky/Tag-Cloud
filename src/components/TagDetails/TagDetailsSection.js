import React, {Component, Fragment} from "react";
import PropTypes from "prop-types";

class TagDetailsSection extends Component {

	generateItems = (labels, data, colors) => {
		const maxValue = Math.max(...labels.map(itemLabel => data[itemLabel]));

		return labels.map((itemLabel, index) => {
			const itemValue = data[itemLabel];

			const lineWidth = Math.floor((itemValue * 100) / maxValue);

			return (
				<Fragment key={itemLabel + index}>
					<p className="tag-details_section_item_label">{itemLabel}</p>
					<div className="tag-details_section_item_graph" style={{width: `${lineWidth}%`, backgroundColor: colors[index]}}></div>
					<p className="tag-details_section_item_value">{itemValue}</p>
				</Fragment>
			);
		});
	}

	render() {
		const {title, data, colors} = this.props;

		const dataLabels = Object.keys(data);

		return (
			<Fragment>
				<h3 className="tag-details_section_title">{title}</h3>
				{this.generateItems(dataLabels, data, colors)}
			</Fragment>
		);
	}
}

TagDetailsSection.propTypes = {
	title: PropTypes.string.isRequired,
	data: PropTypes.object.isRequired,
	colors: PropTypes.array.isRequired
};

export default TagDetailsSection;