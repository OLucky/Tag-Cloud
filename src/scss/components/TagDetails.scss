.tag-details_subwrap {
	display: grid;
	grid-template-columns: auto 1fr auto;
	column-gap: 2rem;
	row-gap: 0.5rem;
	align-items: center;
	max-width: 500px;
}

.tag-details_section_title {
	grid-column: 1 / span 3;
	margin: 1rem 0;
}

.tag-details_section_item {
	display: flex;
	align-items: center;
	max-width: 500px;
}

.tag-details_section_item_graph {
	height: 2rem;
}