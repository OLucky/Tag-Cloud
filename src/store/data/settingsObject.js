export const settingsObject = {
	minFontSize: 16,
	maxFontSize: 160,
	colors: ["#000000", "#0000FF", "#FF00FF",
		"#008000", "#800000",
		"#000080", "#808000", "#800080",
		"#C0C0C0", "#008080", "#FF6347"],
	numberOfPages: 9
};