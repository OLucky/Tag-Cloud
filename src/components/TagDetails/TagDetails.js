import React, {Component, Fragment} from "react";
import {Helmet} from "react-helmet";
import PropTypes from "prop-types";

import TagDetailsSection from "./TagDetailsSection";

class TagDetails extends Component {
	render(){
		const {tag, history, settings} = this.props;

		return(
			<Fragment>
				<Helmet>
					<title>Tag Cloud - {tag.label}</title>
				</Helmet>
				
				<div className="container tag-details">
					<button onClick={history.goBack}>Go Back</button>
					<h1>{tag.label}</h1>
					<h2>Total Mentions - {tag.volume}</h2>
					<div className="tag-details_subwrap">
						<TagDetailsSection title="Mentions per type" data={tag.sentiment} colors={settings.colors} />
						<TagDetailsSection title="Mentions per page" data={tag.pageType} colors={settings.colors} />
					</div>
				</div>
			</Fragment>
		);
	}
}

TagDetails.propTypes = {
	tag: PropTypes.object.isRequired,
	history: PropTypes.object,
	settings: PropTypes.object.isRequired
};

export default TagDetails;