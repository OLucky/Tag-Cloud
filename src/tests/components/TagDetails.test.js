import React from "react";
import {render} from "enzyme";
import toJson from "enzyme-to-json";

import "../config";

import TagDetails from "../../components/TagDetails/TagDetails";

import {tagsArray} from "../../store/data/tagsArray";
import {settingsObject} from "../../store/data/settingsObject";

const oneTag = tagsArray[0];

describe("TagDetails component", () => {

	test("renders correctly with children", () => {
		const mockData = {
			label: oneTag.label,
			id: oneTag.id,
			sentiment: oneTag.sentiment,
			pageType: oneTag.pageType,
			volume: oneTag.volume
		};

		const mockHistory = {
			goBack: jest.fn()
		};

		const component = render(<TagDetails tag={mockData} settings={settingsObject} history={mockHistory}/>);

		expect(toJson(component)).toMatchSnapshot();
	});

});