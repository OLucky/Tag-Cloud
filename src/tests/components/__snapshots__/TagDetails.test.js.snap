// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagDetails component renders correctly with children 1`] = `
<div
  class="container tag-details"
>
  <button>
    Go Back
  </button>
  <h1>
    Odessa
  </h1>
  <h2>
    Total Mentions - 165
  </h2>
  <div
    class="tag-details_subwrap"
  >
    <h3
      class="tag-details_section_title"
    >
      Mentions per type
    </h3>
    <p
      class="tag-details_section_item_label"
    >
      negative
    </p>
    <div
      class="tag-details_section_item_graph"
      style="width:2%;background-color:#000000"
    />
    <p
      class="tag-details_section_item_value"
    >
      3
    </p>
    <p
      class="tag-details_section_item_label"
    >
      neutral
    </p>
    <div
      class="tag-details_section_item_graph"
      style="width:100%;background-color:#0000FF"
    />
    <p
      class="tag-details_section_item_value"
    >
      133
    </p>
    <p
      class="tag-details_section_item_label"
    >
      positive
    </p>
    <div
      class="tag-details_section_item_graph"
      style="width:21%;background-color:#FF00FF"
    />
    <p
      class="tag-details_section_item_value"
    >
      29
    </p>
    <h3
      class="tag-details_section_title"
    >
      Mentions per page
    </h3>
    <p
      class="tag-details_section_item_label"
    >
      blog
    </p>
    <div
      class="tag-details_section_item_graph"
      style="width:30%;background-color:#000000"
    />
    <p
      class="tag-details_section_item_value"
    >
      17
    </p>
    <p
      class="tag-details_section_item_label"
    >
      facebook
    </p>
    <div
      class="tag-details_section_item_graph"
      style="width:100%;background-color:#0000FF"
    />
    <p
      class="tag-details_section_item_value"
    >
      56
    </p>
    <p
      class="tag-details_section_item_label"
    >
      forum
    </p>
    <div
      class="tag-details_section_item_graph"
      style="width:39%;background-color:#FF00FF"
    />
    <p
      class="tag-details_section_item_value"
    >
      22
    </p>
    <p
      class="tag-details_section_item_label"
    >
      general
    </p>
    <div
      class="tag-details_section_item_graph"
      style="width:8%;background-color:#008000"
    />
    <p
      class="tag-details_section_item_value"
    >
      5
    </p>
    <p
      class="tag-details_section_item_label"
    >
      image
    </p>
    <div
      class="tag-details_section_item_graph"
      style="width:0%;background-color:#800000"
    />
    <p
      class="tag-details_section_item_value"
    >
      0
    </p>
    <p
      class="tag-details_section_item_label"
    >
      news
    </p>
    <div
      class="tag-details_section_item_graph"
      style="width:46%;background-color:#000080"
    />
    <p
      class="tag-details_section_item_value"
    >
      26
    </p>
    <p
      class="tag-details_section_item_label"
    >
      review
    </p>
    <div
      class="tag-details_section_item_graph"
      style="width:1%;background-color:#808000"
    />
    <p
      class="tag-details_section_item_value"
    >
      1
    </p>
    <p
      class="tag-details_section_item_label"
    >
      twitter
    </p>
    <div
      class="tag-details_section_item_graph"
      style="width:62%;background-color:#800080"
    />
    <p
      class="tag-details_section_item_value"
    >
      35
    </p>
    <p
      class="tag-details_section_item_label"
    >
      video
    </p>
    <div
      class="tag-details_section_item_graph"
      style="width:5%;background-color:#C0C0C0"
    />
    <p
      class="tag-details_section_item_value"
    >
      3
    </p>
  </div>
</div>
`;
